src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -125,10 +125,15 @@ public function registerClasses(Definition $prototype, string $namespace, string
         foreach ($classes as $class => $errorMessage) {
             if (null === $errorMessage && $autoconfigureAttributes) {
                 $r = $this->container->getReflectionClass($class);
-                if ($r->getAttributes(Exclude::class)[0] ?? null) {
-                    $this->addContainerExcludedTag($class, $source);
-                    continue;
+                $exclusionAttributes = $this->container->getExclusionAttributes();
```

Comment:
IIRC, file loaders are run on separate container instances, right? This would mean that a bundle cannot expect registerAttributeForExclusion to have any effect on classes in `src/`. Am I correct? If yes, don't we want to make these app-wide instead?

Reply:
what about this? did you give it a try? configuring an exclusion within a bundle and seeing it work on the app?